docs/csharp/versioning.md

```diff
@@ -0,0 +1,128 @@
+
+## Consuming Libraries
+
+As a developer that consumes .NET libraries built by other developers you're most likely aware that a new version of a library might not be fully compatible with your project
+and you might often find yourself having to update your code to work with those changes.
+
+Lucky for you C# and the .NET ecosystem comes with features and techniques that allow us to easily update our app to work with new versions of libraries that might introduce breaking changes.
+
+### Assembly Binding Redirection
+
+You can use the `app.config` file to update the version of a library your app uses. By adding what is called a `binding redirect` your
+can use the new library version without having to recompile your app. The following example shows how you would update
+your app's `app.config` file to use the `1.0.1` patch version of `ReferencedLibrary` instead of the `1.0.0` version it was originally compiled with.
+
+```xml
+<dependentAssembly>
+    <assemblyIdentity name="ReferencedLibrary" publicKeyToken="32ab4ba45e0a69a1" culture="en-us" />
+    <bindingRedirect oldVersion="1.0.0" newVersion="1.0.1" />
+</dependentAssembly>
+```
+
+Go [here](https://msdn.microsoft.com/en-us/library/7wd6ex19(v=vs.110).aspx) to get more information on redirecting assembly versions.
+
+### new
+
+You use the `new` modifier to hide inherited members of a base class. This ensures resiliency of derived classes allowing you to update base classes the way you want in your library.
+
+Take the following example:
+
+```
+[!code-csharp[Sample usage of the 'new' modifier](/samples/csharp/versioning/new/Program.cs#sample)]
+```
+
+#### Output
+
+```
+A base method
+A derived method
+```
+
+From the example above you can see how `DerivedClass` hides the `MyMethod` method present in `BaseClass`.
+This means that when a base class in the new version of a library adds a member that already exists in your derived class, you can
+simply use the `new` modifier on your derived class member to hide the base class member.
+
+When no `new` modifier is specified, a derived class will by default hide conflicting members in a base class,
+although a compiler warning will be generated the code will still compile. This means that simply adding new members to an existing class
+makes that new version of your library both source and binary compatible with code that depends on it.
+
+### override
+
+The `override` modifier works very similar to the `new` modifier except that it extends the implementation of a base class member rather than
+hides it, also the base class member needs to have the `virtual` modifier applied to it.
+
+```
+[!code-csharp[Sample usage of the 'override' modifier](/samples/csharp/versioning/override/Program.cs#sample)]
+```
+
+#### Output
+
+```
+Base Method One: Method One
+Derived Method One: Derived Method One
+```
+
+The `override` modifier is evaluated at compile time as opposed to at runtime for the `new` modifier, and the compiler will throw an error
```

Comment:
I don't understand what is this trying to say. How is `new` evaluated at runtime?